codec/reflectcodec/type_codec.go

```diff
@@ -113,12 +118,22 @@ func (c *genericCodec) size(value reflect.Value) (int, bool, error) {
 	case reflect.String:
 		return wrappers.StringLen(value.String()), false, nil
 	case reflect.Ptr:
-		if value.IsNil() {
-			// Can't marshal nil pointers (but nil slices are fine)
+		if omitEmpty {
+			if value.IsNil() {
+				return wrappers.BoolLen, false, nil
+			}
+			size, _, err := c.size(value.Elem(), omitEmpty)
+			if err != nil {
+				return 0, false, err
+			}
+			return wrappers.BoolLen + size, false, nil
+		}
+
+		if value.IsNil() { // Can't marshal nil (except nil slices *or* if omitEmpty)
 			return 0, false, errMarshalNil
 		}
-		return c.size(value.Elem())
 
+		return c.size(value.Elem(), omitEmpty)
 	case reflect.Interface:
```

Comment:
Do we want to add `nil` support for interfaces?

Reply:
I think so, I'll make it happen.